tests/testthat/test-get_custom_effort.R

```diff
@@ -91,60 +141,62 @@ test_that("right columns, cols types, right relative number of rows", {
 
   # right cols and col types: group by month
   effort_by_month <- get_custom_effort(mica, group_by = "month")
-  expect_true(
-    all(colnames(effort_by_month) == c("begin", "effort", "unit"))
-  )
+  expect_named(effort_by_month, expected = c("begin", "effort", "unit"))
 
   # right cols and col types: group by week
   effort_by_week <- get_custom_effort(mica, group_by = "week")
-  expect_true(
-    all(
-      colnames(effort_by_week) == c("begin", "effort", "unit")
-    )
-  )
+  expect_named(effort_by_week, expected = c("begin", "effort", "unit"))
 
   # right cols and col types: group by day
   effort_by_day <- get_custom_effort(mica, group_by = "day")
-  expect_true(
-    all(
-      colnames(effort_by_day) == c(
-        "begin",
-        "effort",
-        "unit"
-      )
-    )
-  )
+  expect_named(effort_by_day, expected = c("begin", "effort", "unit"))
 
   # number of rows is equal to 1 if group_by is NULL
-  expect_equal(nrow(tot_effort), 1)
+  expect_identical(nrow(tot_effort), 1L)
 
-  # number of rows with grouping by year is equal to number of days divided by
-  # 365
+  # number of rows with grouping by year is equal to number of calendar years
   first_day <- min(mica$data$deployments$start)
   last_day <- max(mica$data$deployments$end)
-  n_years <- as.numeric(last_day - first_day) %/% 365 + 1
-  expect_equal(nrow(effort_by_year), n_years)
+  n_years <- length(seq(
+    lubridate::floor_date(first_day, unit = "years"), 
+    lubridate::floor_date(last_day, unit = "years"),
+    by = "years")
+  )
+  expect_identical(nrow(effort_by_year), n_years)
 
-  # number of rows with grouping by month is equal to number of days divided by
-  # 30
-  n_months <- as.numeric(last_day - first_day) %/% 30 + 1
-  expect_equal(nrow(effort_by_month), n_months)
+  # number of rows with grouping by month is equal to number of calendar months
+  n_months <- length(seq(
+    lubridate::floor_date(first_day, unit = "months"), 
+    lubridate::floor_date(last_day, unit = "months"),
+    by = "months")
+  )
+  expect_identical(nrow(effort_by_month), n_months)
 
-  # number of rows with grouping by week is equal to number of days divided by 7
-  n_weeks <- as.numeric(last_day - first_day) %/% 7 + 1
-  expect_equal(nrow(effort_by_week), n_weeks)
+  # number of rows with grouping by week is equal to number of calendar weeks
+  n_weeks <- length(seq(
+    lubridate::floor_date(first_day, unit = "weeks"), 
```

Comment:
```suggestion
    lubridate::floor_date(first_day, unit = "weeks"),
```
remove trailing whitespace